delivery_package_number/models/stock_picking.py

```diff
@@ -18,11 +18,12 @@ class StockPicking(models.Model):
     )
     ask_number_of_packages = fields.Boolean(compute="_compute_ask_number_of_packages")
 
-    @api.depends("package_ids")
+    @api.depends("move_line_ids", "move_line_ids.result_package_id")
     def _compute_number_of_packages(self):
         for picking in self:
-            if picking.package_ids:
-                picking.number_of_packages = len(picking.package_ids)
+            packages = picking.move_line_ids.result_package_id
+            if packages:
```

Comment:
why this `if` ?

Reply:
Because we don't want to loose the info set on `number_of_packages` when no packages are set when modifying the move_line_ids